src/USER-BOCS/fix_bocs.cpp

```diff
@@ -630,154 +635,194 @@ void FixBocs::init()
-        data[1][n_entries-1] = (double) f2;
-        if (n_entries == 2) {
-          stdVolumeInterval = data[0][n_entries-1] - data[0][n_entries-2];
+        //message = fmt::format("INFO: f1 = {}, f2 = {}", f1, f2);
+        //error->message(FLERR,message);
+        data[VOLUME][i] = (double)f1;
+        data[PRESSURE_CORRECTION][i] = (double)f2;
+        if (i == 1)
+        {
+          // second entry is used to compute the validation interval used below
+          stdVolumeInterval = data[VOLUME][i] - data[VOLUME][i-1];
+          //message = fmt::format("INFO: standard volume interval computed: {}", stdVolumeInterval);
+          //error->message(FLERR,message);
         }
-        else if (n_entries > 2) {
-          currVolumeInterval = data[0][n_entries-1] - data[0][n_entries-2];
+        else if (i > 1)
+        {
+          // after second entry, all intervals are validated
+          currVolumeInterval = data[VOLUME][i] - data[VOLUME][i-1];
+          //message = fmt::format("INFO: current volume interval: {}", currVolumeInterval);
+          //error->message(FLERR,message);
           if (fabs(currVolumeInterval - stdVolumeInterval) > volumeIntervalTolerance) {
-            snprintf(badDataMsg,MAX_MESSAGE_LENGTH,
-                     "BAD VOLUME INTERVAL: spline analysis requires uniform"
-                     " volume distribution, found inconsistent volume"
-                     " differential, line %d of file %s\n\tline: %s",
-                     n_entries,filename,line);
-            error->message(FLERR,badDataMsg);
+            message = fmt::format("ERROR: Bad volume interval. Spline analysis requires uniform"
+                    " volume distribution, found inconsistent volume"
+                    " differential, line {} of file {}\n\tline: {}",
+                    lineNum,filename,inputLines.at(i));
+            error->message(FLERR,message);
             badInput = true;
+            numBadVolumeIntervals++;
           }
+          // no concluding else is intentional: i = 0, first line, no interval to validate
         }
-        // no else -- first entry is simply ignored
       }
       else
       {
-        snprintf(badDataMsg,MAX_MESSAGE_LENGTH,
-                 "BAD INPUT FORMAT: did not find 2 comma separated numeric"
-                 " values in line %d of file %s\n\tline: %s",
-                 n_entries,filename,line);
-        error->message(FLERR,badDataMsg);
+        message = fmt::format("ERROR: Bad input format: did not find 2 comma separated numeric"
+                 " values in line {} of file {}\n\tline: {}",
+                 lineNum, filename, inputLines.at(i));
+        error->message(FLERR,message);
         badInput = true;
       }
+      if (badInput)
+      {
+        numBadVolumeIntervals++;
+      }
+    }
+
+    if (numBadVolumeIntervals) {
+      message = fmt::format("INFO: total number bad volume intervals = {}", numBadVolumeIntervals);
+      error->message(FLERR,message);
```

Comment:
please see my other comments about messages printed to the screen. In this case, actually removing the "INFO" tag and using message->warning() would be more appropriate.